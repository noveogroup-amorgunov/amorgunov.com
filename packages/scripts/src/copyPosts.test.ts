import fs from 'node:fs'
import { afterEach, describe, expect, test, vi } from 'vitest'
import { copyPosts } from './copyPosts.js'

describe('copyPosts', () => {
  afterEach(() => {
    vi.resetModules()
  })

  test('should copy post with updating images path', async () => {
    const spy = vi.spyOn(fs, 'writeFileSync')

    spy.mockImplementationOnce(() => 'calling mock function')

    const targetDir = 'src/__tests__/'
    const destinationDir = 'src/__tests__/'

    await copyPosts(targetDir, destinationDir)

    expect(spy).toHaveBeenCalledTimes(1)
    expect(spy).toMatchSnapshot()
  })
})
