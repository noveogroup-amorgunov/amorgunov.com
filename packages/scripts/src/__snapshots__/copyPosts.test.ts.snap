// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`copyPosts > should copy post with updating images path 1`] = `
[MockFunction writeFileSync] {
  "calls": [
    [
      "src/__tests__/2025-05-18-test-slug.md",
      "---
title: "test-post"
date: 2025-05-18
time: 5
description: "ХХХ"
featuredImageThumbnail: "/assets/images/2025-05-18-test-slug/preview.jpg"
tags:
  - ХХХ
  - ХХХ
layout: layouts/post.hbs
likes: 0
---

# Test post

<img
  class="lazyload inverting"
  alt="зацепление против связанности"
  src="/assets/images/2025-05-20-how-solve-cross-imports/CouplingVsCohesion.min.png"
  data-src="/assets/images/2025-05-20-how-solve-cross-imports/CouplingVsCohesion.out.png"
/>

## Second image

<img
  class="lazyload inverting"
  alt="зацепление против связанности"
  src="/assets/images/2025-05-20-how-solve-cross-imports/CouplingVsCohesion.png"
  data-src="/assets/images/2025-05-20-how-solve-cross-imports/CouplingVsCohesion.png"
/>
",
      "utf-8",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": "calling mock function",
    },
  ],
}
`;
